roles/pulp_health_check/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+---
+- meta: flush_handlers
+
+- name: Get services
+  service_facts:
+
+- name: Listing Pulp services
+  debug:
+    msg: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+
+- name: Checking Pulp services
+  fail:
+    msg: "{{ item.value.name }} state: {{ item.value.state }}"
+  when: "item.value.name != 'pulpcore-worker@.service' and item.value.state != 'running'"
+  with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+- name: Ensure Pulp is up and healthy
+  uri:
+    url: "http://{{ pulp_api_bind }}/pulp/api/v3/status/"
```

Comment:
Maybe it would be a good idea to also provide a way to select the default API schema (either HTTP or HTTPS)?

Reply:
So this role is exclusively meant to run against the pulp_api host at this time. Not the webserver. It is even meant to be run before the webserver is installed.

So later on once we broaden it for more use cases, we should do that.

We have other design issues too. We use `pulp_api_bind`, but we what if users set that to `0.0.0.0:24817`? One cannot connect to 0.0.0.0, but one can bind to it. My proposal is for the next release to address that by using a different variable like `pulp_health_check_api_origin` that would merely default to `http://{{ pulp_api_bind }}'